### DISCUSSION TITLE
Not able to build gstreamer1.0-plugins-nveglgles_1.2.3-r32.6.1

### DISCUSSION THREAD

**Discussion by matthew-rco** [OE4T/meta-tegra#818] [General]
I am trying to build an image based on yocto dunfell and branch dunfell of meta-tegra (r32.6.1) but I have the following error: 

`| In file included from ../../../gstegl_src/gst-egl/ext/eglgles/gstegladaptation_egl.c:28:
| ../../../gstegl_src/gst-egl/ext/eglgles/gsteglglessink.h:35:10: fatal error: cuda.h: No such file or directory
|  #include <cuda.h>
|           ^~~~~~~~
| compilation terminated.
| In file included from ../../../gstegl_src/gst-egl/ext/eglgles/gsteglglessink.c:121:
| ../../../gstegl_src/gst-egl/ext/eglgles/gsteglglessink.h:35:10: fatal error: cuda.h: No such file or directory
|  #include <cuda.h>
|           ^~~~~~~~
| compilation terminated.
`

In my package list I already have cuda:
cuda-cudart
cuda-driver
cuda-libraries
cuda-nvcc-headers
cuda-nvgraph
cuda-nvml
cuda-nvrtc
cuda-samples
cuda-target-environment

Do you have any idea what is missing ?

**Comment by ichergui** [comment]
Hello @matthew-rco 

Are you using [tegra-demo-distro](https://github.com/OE4T/tegra-demo-distro) ? if so, this should be working OK.
Please make sure that you updated the `tegra-demo-distro` becauce we introduce [meta-tegra-community](https://github.com/OE4T/meta-tegra-community) recently.

Let me know if your issue persist

**Comment by matthew-rco** [comment]
Hi @ichergui,

I am updating from branch dunfell-l4t-r32.4.2 so I already have poky branch dunfell and meta-openembedded branch dunfell and all I want to change is layer meta-tegra to branch dunfell (r32.6.1). I am building again after doing a pull of poky on branch dunfell and pull of meta-openembedded branch dunfell, maybe this was causing the issue. I will let you know. 

Is the layer meta-tegra-community required for build ?

**Reply by ichergui** [reply]
When using `tegra-demo-distro`, the `meta-tegra-community` meta layer is included in `bblayers.conf`
So it is mandatory.

**Reply by matthew-rco** [reply]
I am not using tegra-demo-distro, only meta-tegra and meta-openembedded layers with poky.

**Comment by madisongh** [comment]
What target MACHINE are you building for? Do you see `cuda-cudart` in the recipe dependencies, e.g.:
```
$ bitbake -e gstreamer1.0-plugins-nveglgles | grep '^DEPENDS='
DEPENDS="pkgconfig-native autoconf-native automake-native libtool-native libtool-cross gnu-config-native  virtual/aarch64-oe4t-linux-gcc virtual/aarch64-oe4t-linux-compilerlibs virtual/libc gettext-native gstreamer1.0 glib-2.0-native gstreamer1.0-plugins-base virtual/egl virtual/libgles2 cuda-cudart tegra-libraries python3-native  gobject-introspection gobject-introspection-native qemu-native prelink-native wayland libx11 libxext"
```
and if you look at the `log.do_compile`, are you seeing `...recipe-sysroot/usr/local/cuda-10.2/include` as one of the include paths passed to `libtool --mode=compile` ?

**Comment by matthew-rco** [comment]
It works with the last version of poky dunfell and meta-openembedded dunfell pulled from git. I was at old versions of the repos, my mistake.
Thank you for your help.

**Reply by ichergui** [reply]
Perfect 
Thanks @matthew-rco for the update
